apps/emqx_s3/src/emqx_s3_client.erl

```diff
@@ -135,12 +137,14 @@ start_multipart(
         {error, Reason} ->
             ?SLOG(debug, #{msg => "start_multipart_fail", key => Key, reason => Reason}),
             {error, Reason}
-    end.
+    end;
+start_multipart(#{aws_config := {error, Reason}}, _Key, _UploadOpts) ->
```

Comment:
Curious: what is the advantage of making `{error, ...}` a possible value for the call?
(Against checking on the caller side)

Reply:
Honestly, just wanted to make sure that `emqx_s3` API changes in this PR are minimal, which in turn led to this strange situation where it's possible to create a client with broken config. I'm not really against moving the responsibility to a caller, open to suggestions on how to make it prettier.